app/app.py

```diff
@@ -85,49 +88,69 @@ def extract_signature_info(image_path):
+    name_address_combo = f"{name} {address}"
+    # Searches for a match within the Ward returned by OCR
+    high_match_ids = score_fuzzy_match_slim(name_address_combo, voter_records_2023_df[voter_records_2023_df['WARD'] == f"{dict_['Ward']}.0"]["OCR"])
+    name_, score_, id_ = high_match_ids[0]
+    # If no Valid matches are found, searches for a match against the entire registry except the ward already checked
+    if score_ < 85.0:
+        high_match_ids = score_fuzzy_match_slim(name_address_combo, voter_records_2023_df[voter_records_2023_df['WARD'] != f"{dict_['Ward']}.0"]["OCR"])
+        if score_ < high_match_ids[0][1]:
+            name_, score_, id_ = high_match_ids[0]
+    if score_ >= 85.0:
+        return high_match_ids[0]
+    # IF no Valid matches have been found, searches for a match using only the Full Name
+    else:
+        matched_full_names = score_fuzzy_match_slim(name, voter_records_2023_df["Full Name"], scorer_=fuzz.ratio)
+        full_name, full_name_score, full_name_id = matched_full_names[0]
+    # Compare scores of full name + address match to score of Full Name match and take the record with the highest score in the format Tuple(matched_record, score, index)
+    if score_ > full_name_score:
+        return high_match_ids[0]
+    else:
+        address = voter_records_2023_df.loc[full_name_id, 'Full Address']
+        full_name = f"{full_name} {address}"
+        return (full_name, full_name_score, full_name_id)
 
+##
+# DELETE TEMPORARY FILES
+##
 
+def wipe_temp_dir_status_bar(remove_status_bar):
+    index = 0
+    pattern = os.path.join('.', 'temp_ocr_images', '*')
+    temp_files = glob.glob(pattern)
+    for file in temp_files:
+        os.remove(file)
+        remove_status_bar.progress((index+1)/len(temp_files), text="Temporary Image Files Removed")
+        index += 1
+
+def wipe_temp_dir():
+    pattern = os.path.join('.', 'temp_ocr_images', '*')
+    temp_files = glob.glob(pattern)
+    for file in temp_files:
+        os.remove(file)
 ##
 # DATA UPLOAD AND FULL NAME GENERATION
 ##
 
 # reading in election data
-voter_records_2023_df = pd.read_csv('data/raw_feb_23_city_wide.csv', dtype=str)
-
-# creating full name column
-voter_records_2023_df['Full Name'] = voter_records_2023_df.apply(lambda x: f"{x['First_Name']} {x['Last_Name']}", axis=1)
-full_name_list = list(voter_records_2023_df['Full Name'])
 
+voter_records_2023_df = pd.read_csv('data/raw_feb_23_city_wide.csv', dtype=str)
 
+# creating full name, full address, and OCR columns
+voter_records_2023_df['Full Name'] = voter_records_2023_df["First_Name"] + ' ' + voter_records_2023_df['Last_Name']
+voter_records_2023_df['Full Address'] =  voter_records_2023_df["Street_Number"] + " " + voter_records_2023_df["Street_Name"] + " " + voter_records_2023_df["Street_Type"] + " " + voter_records_2023_df["Street_Dir_Suffix"]
+voter_records_2023_df['OCR'] = voter_records_2023_df["Full Name"] + ' ' + voter_records_2023_df["Full Address"]
```

Comment:
The 'OCR' column should be called 'Full Name and Full Address'. Calling it 'OCR' is confusing since this column isn't derived from OCR.